src/utils/crt_launch/crt_launch.c

```diff
@@ -131,13 +142,22 @@ parse_args(int argc, char **argv)
 
 /* Retrieve self uri via CART */
 static int
-get_self_uri(struct host *h)
+get_self_uri(struct host *h, int rank)
 {
 	char		*uri;
 	crt_context_t	ctx;
 	char		*p;
 	int		len;
 	int		rc;
+	char		*str_port;
```

Comment:
Shouldn't there be a test here to make sure rank is between 0 and 99?

Reply:
Once we fix this issue which is blocking/urgent right now, I will add extra verifications into crt_launch, namely it will parse out contents of the reserved range file and report warning if rank+port falls outside of the reserved range on that system.

Reply:
OK